src/group_theory/exponent.lean

```diff
@@ -150,11 +200,69 @@ begin
     simp only [true_and, set.mem_univ, finset.coe_univ] at hg,
     exact ne_of_gt (order_of_pos g) hg },
   { intro g,
-    have h : (order_of g) ∣ (finset.univ : finset H).lcm order_of,
+    have h : (order_of g) ∣ (finset.univ : finset G).lcm order_of,
     { apply finset.dvd_lcm,
       exact finset.mem_univ g },
     cases h with m hm,
     rw [hm, pow_mul, pow_order_of_eq_one, one_pow] },
 end
 
+@[to_additive]
+lemma exponent_ne_zero_of_fintype [fintype G] : exponent G ≠ 0 :=
+by simpa [←lcm_order_eq_exponent, finset.lcm_eq_zero_iff] using λ x, (order_of_pos x).ne'
+
+end left_cancel_monoid
+
+section comm_monoid
+
+variable [cancel_comm_monoid G]
+
+open finset
+
+-- TODO: this is true without the `fintype`.
```

Comment:
How hard is it to implement this?

Reply:
d'oh, this doesn't hold. set.range (order_of : ℤ → ℕ) = {0,1,2}, but ℤ's exponent is zero. I think you need all members being of positive order, and then it holds. I'll try for a bit to prove this.

Reply:
here's a version with one sorry: https://gist.github.com/ericrbg/26bee67ca48963020bb1b8a407503c0a. it's not a hard sorry but my brain is frazzled right now.

Reply:
I finished this in the other PR. I'll merge it in when everything is in master